### ISSUE TITLE
pseudo-class nested in a media query does not output anything, the reverse does.

### ISSUE THREAD

**Issue by Stijnkool** [rtfeldman/elm-css#352]
This works (outputs the correct css):
```elm
div
    [ css
        [ nthOfType "2n+1"
            [ withMedia [ only screen [ Media.minWidth (px 600) ] ]
                [ marginRight (px 16) ]
            ]
        ]
    ]
    []
```
This does not work, but is allowed (does not output any css):
```elm
div
    [ css
        [ withMedia [ only screen [ Media.minWidth (px 600) ] ]
            [ nthOfType "2n+1"
                [ marginRight (px 16) ]
            ]
        ]
    ]
    []
```
This is also the way you write normal css, with the media query on the outside.

- Because we have to use the media query on the inside of the pseudo-class, you have to write multiple media queries if you want extra rules not related to the pseudo class. This results in extra generated media queries of the same kind. These are not combined in the output. But I suspect they don't need to be combined if you write only 1 per query.

**Comment by rtfeldman** [comment]
@tolgap any ideas what the culprit might be? 🤔

**Comment by edkv** [comment]
This is a duplicate of #311.

**Comment by tolgap** [comment]
@rtfeldman Yup I know where to find the culprit, will take a crack at it this week.

**Comment by tolgap** [comment]
@rtfeldman @edkv I've looked into this a few hours. I have found the culprit, but the solution is mind boggling.

The issue is right at this line: https://github.com/rtfeldman/elm-css/blob/master/src/Css/Preprocess/Resolve.elm#L381

On that line, we lose all nested children's information with regards to the media query.

That is because `collectSelectors` does not collect any child selectors: https://github.com/rtfeldman/elm-css/blob/master/src/Css/Preprocess/Resolve.elm#L519

I've tried, but couldn't quite work it out without creating all kinds of spaghetti code. Might take a shot at it some other time, but in the mean time if anyone else wants to take a shot at it: hit me up and I'll gladly help 👍 .
